src/org/intellij/erlang/psi/impl/ErlangPsiImplUtil.java

```diff
@@ -1821,7 +1864,22 @@ public static boolean fromTheSameCaseExpression(@NotNull PsiElement origin, @Not
     return false;
   }
 
-  public static class ErlangFunctionCallParameter<T extends PsiElement> extends PatternCondition<T> {
+  public static abstract class ErlangFunctionCallParameterBase<T extends PsiElement> extends PatternCondition<T> {
+    protected ErlangFunctionCallExpression myCall;
+    protected int myPositionInCall;
```

Comment:
In that case I don't see advantages of having a superclass.
That 3 lines in all classes initializes a state which is used below in the code.